flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistryImpl.java

```diff
@@ -113,19 +114,12 @@ public MetricRegistryImpl(MetricRegistryConfiguration config) {
 			// by default, don't report anything
 			LOG.info("No metrics reporter configured, no metrics will be exposed/reported.");
 		} else {
-			// we have some reporters so
-			for (Tuple2<String, Configuration> reporterConfiguration: reporterConfigurations) {
-				String namedReporter = reporterConfiguration.f0;
-				Configuration reporterConfig = reporterConfiguration.f1;
-
-				final String className = reporterConfig.getString(ConfigConstants.METRICS_REPORTER_CLASS_SUFFIX, null);
-				if (className == null) {
-					LOG.error("No reporter class set for reporter " + namedReporter + ". Metrics might not be exposed/reported.");
-					continue;
-				}
+			for (ReporterSetup reporterSetup : reporterConfigurations) {
+				final String namedReporter = reporterSetup.getName();
+				final MetricConfig metricConfig = reporterSetup.getConfiguration();
 
 				try {
-					String configuredPeriod = reporterConfig.getString(ConfigConstants.METRICS_REPORTER_INTERVAL_SUFFIX, null);
+					String configuredPeriod = metricConfig.getString(ConfigConstants.METRICS_REPORTER_INTERVAL_SUFFIX, null);
```

Comment:
Should we hide the configuration field accesses behind `MetricConfig#getReporterInterval()`?

Reply:
I don't really see the benefit though

Reply:
I don't really see the benefit, nor the difference to any access to a `Configuration` where we have not introduced such utility method so far _in general_.

In any case I wouldn't do it in this PR at the very least.

Reply:
The benefit would be that we hide the fact that we read the value from the underlying `Configuration`. It would, thus, follow better the law of Demeter.

Reply:
I may be biased here, but I don't believe we're hiding anything. `MetricConfig` is just a minor extension of `Properties` (which in hindsight shouldn't even exist) and I have no plans to change that. So long as that is the case there's no other place the value could come from but the configuration.

As a more relatable example, would you say that `Configuration#getJobManagerHost()` hides in any way that the value is read from the configuration?

Reply:
I think in general it is better to offer a restricted API because it usually simplifies things. Here we have to provide the correct key in order to find the right value. From a user's perspective who does not know the details of how things are stored in `MetricConfig` it would be easier to simply call `MetricConfig#getReporterInterval`. It would even be easier to call `ScheduledMetricReporter#getReportertInterval` to relate this discussion to my other point.